runtime/port/unix/j9sysinfo.c

```diff
@@ -566,6 +610,62 @@ mapPPCProcessor(const char *processorName)
 }
 #endif /* defined(LINUXPPC) */
 
+/**
+ * @internal
+ * Reads the content of the file to the array
+ *
+ * @param[in] metricKeyInFile - pointer to the string which holds key in Cgroup file
+ *
+ * @param[in] file - pointer to the Cgroup Subsystem file
+ *
+ * @param[in] value - Pointer to the char to store the content of file
+ *
+ * returns 0 on success J9PORT_ERROR_SYSINFO_CGROUP_SUBSYSTEM_UNAVAILABLE on failure
+ */
+#if defined(LINUX) && !defined(J9ZTPF)
+static int32_t
+readCgroupMetricFromFile(struct J9PortLibrary *portLibrary, char *metricKeyInFile, FILE *file, char *value)
+{
+	int32_t rc = J9PORT_ERROR_SYSINFO_CGROUP_SUBSYSTEM_UNAVAILABLE;
+	if (file == NULL) {
+		goto _end;
+	}
+	if (metricKeyInFile == NULL) {
+		if (fgets(value, MAX_LINE_LENGTH, file) != NULL) {
+			rc = 0;
+			goto _end;
+		}
+	} else {
+		char bufferStore[MAX_LINE_LENGTH] = {0};
+		char *tmpPtr = NULL;
+		while (0 == feof(file)) {
+			if (NULL != fgets(bufferStore, MAX_LINE_LENGTH, file)) {
+				size_t size = strlen(metricKeyInFile);
+				tmpPtr = (char *)bufferStore;
```

Comment:
This cast should not be necessary; please remove it. The declaration of `tmpPtr` should move here.